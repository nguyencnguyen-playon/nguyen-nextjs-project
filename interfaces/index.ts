export interface IPost {
    id?: number;
    name: string;
    desc: string;
    image_url: string;
}

export type PostFormType = {
    name?: string;
    desc?: string;
};