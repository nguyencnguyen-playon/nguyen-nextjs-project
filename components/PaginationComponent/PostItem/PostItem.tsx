'use client'
import Image from 'next/image'
import styled from 'styled-components';
import { Card } from 'antd';
import { IPost } from '@/interfaces';

const { Meta } = Card;

const Container = styled.div`
    display: flex;
    flex-direction: column;
    background: #FFFFFF;
    border-radius: 8px;
    box-shadow: 0px 4px 8px rgba(0, 0, 0, 0.1);
    margin: 16px;
`

export const PostItem = ({ item }: { item: IPost }) => (
    <Container>
        <Card
            hoverable
            style={{ width: 240 }}
            cover={<Image alt="example" width={240} height={240} src={item.image_url || 'https://via.placeholder.com/240x240'} />}
        >
            <Meta title={item.name} description={item.desc || "Lorem ipsum"} />
        </Card>
    </Container>
)