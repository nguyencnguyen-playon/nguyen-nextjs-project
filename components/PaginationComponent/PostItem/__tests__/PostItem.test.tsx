import { render } from '@testing-library/react';
import { PostItem } from '../PostItem';
import { IPost } from '@/interfaces';

describe('PostItem', () => {
    const mockPost: IPost = {
        name: 'Test Post',
        desc: 'Lorem ipsum',
        image_url: 'https://example.com/image.jpg',
    };

    const mockPostNoImage: IPost = {
        name: 'Test Post',
        desc: 'Lorem ipsum',
        image_url: '',
    };

    it('renders the post name and description', () => {
        const component = render(<PostItem item={mockPost} />);
        expect(component).toMatchSnapshot();
    });

    it('renders the post name and description', () => {
        const component = render(<PostItem item={mockPostNoImage} />);
        expect(component).toBeDefined();
    });
});
