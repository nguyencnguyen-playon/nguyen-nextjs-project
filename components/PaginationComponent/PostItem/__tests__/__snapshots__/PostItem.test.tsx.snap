// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostItem renders the post name and description 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-beqWNU eEsbyz"
      >
        <div
          class="ant-card ant-card-bordered ant-card-hoverable css-dev-only-do-not-override-1okl62o"
          style="width: 240px;"
        >
          <div
            class="ant-card-cover"
          >
            <img
              alt="example"
              data-nimg="1"
              decoding="async"
              height="240"
              loading="lazy"
              src="/_next/image?url=https%3A%2F%2Fexample.com%2Fimage.jpg&w=640&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fexample.com%2Fimage.jpg&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fexample.com%2Fimage.jpg&w=640&q=75 2x"
              style="color: transparent;"
              width="240"
            />
          </div>
          <div
            class="ant-card-body"
          >
            <div
              class="ant-card-meta"
            >
              <div
                class="ant-card-meta-detail"
              >
                <div
                  class="ant-card-meta-title"
                >
                  Test Post
                </div>
                <div
                  class="ant-card-meta-description"
                >
                  Lorem ipsum
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-beqWNU eEsbyz"
    >
      <div
        class="ant-card ant-card-bordered ant-card-hoverable css-dev-only-do-not-override-1okl62o"
        style="width: 240px;"
      >
        <div
          class="ant-card-cover"
        >
          <img
            alt="example"
            data-nimg="1"
            decoding="async"
            height="240"
            loading="lazy"
            src="/_next/image?url=https%3A%2F%2Fexample.com%2Fimage.jpg&w=640&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fexample.com%2Fimage.jpg&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fexample.com%2Fimage.jpg&w=640&q=75 2x"
            style="color: transparent;"
            width="240"
          />
        </div>
        <div
          class="ant-card-body"
        >
          <div
            class="ant-card-meta"
          >
            <div
              class="ant-card-meta-detail"
            >
              <div
                class="ant-card-meta-title"
              >
                Test Post
              </div>
              <div
                class="ant-card-meta-description"
              >
                Lorem ipsum
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
